sql/catalyst/src/main/scala/org/apache/spark/sql/catalog/v2/expressions/expressions.scala

```diff
@@ -0,0 +1,208 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.catalog.v2.expressions
```

Comment:
in general i think we should not put internal classes in the same package as public interfaces, especially now we are using Java to define interfaces.

package level visibility gets elided in bytecode, so from library developers' perspective they will just see LogicalExpressions together with all the other public interfaces.

Reply:
I'll fix this when I hear back from you on where to move the public parts of this API.

Reply:
@rxin, where do you want these classes to go?

Reply:
Nevermind, I see a response on a different comment above. It still isn't clear where these should move to and it sounds like the suggestion to reorganize is broader than just the expressions.

If we need to move `expressions`, then let's decide where to move it to. If we need to reorganize `catalog.v2`, then we should do that in a separate commit. I would be fine also getting this in and fixing packages in a follow-up.